### DISCUSSION TITLE
Do we need to register the comment characters used in the "hashpipe" for every new knitr engines?

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#4092] [Q&A] [answered]
Related to #1737

`prql` code block that executes [PRQL](https://prql-lang.org/) queries (like SQL queries) is now available from knitr.
https://eitsupi.quarto.pub/use-prql-executable-code-blocks-on-quarto/

Perhaps because PRQL comments start with `#`, it seems that we can use the hashpipe `#|` without doing anything at the moment.
But to be on the safe side, should I register this in the Quarto source code?

https://github.com/quarto-dev/quarto-cli/blob/56da834f07f5fdfab1e432f11aa3be6b26f4fd2a/src/core/lib/partition-cell-options.ts#L279-L322

Perhaps using `#` for unknown language code blocks now?

**Comment by cscheid** [comment, marked as answer]
Thanks for the suggestion. `kLangCommentChars` so that quarto knows how to extract "comment-pipe" yaml blocks from code cells. As you have noticed, knitr does its own detection of comment-pipe blocks, and so in the knitr engine, this works.

However, if you want IDE support for autocompletion and validation of those cells, quarto needs to know the correct language.

We choose not to use `#|` as a default because we don't want people to get in the habit of writing invalid source code in code cells (that would mean that, for example, our jupyter cell conversion would get more complicated.)

So, the best course of action is for folks to submit PRs with the comment syntax for their languages.

**Reply by eitsupi** [reply]
Thanks, that makes sense.
I'll send a PR.

**Reply by eitsupi** [reply]
Hi, I noticed a similar list in the file below. Should I update this file too?
https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/filters/quarto-pre/engine-escape.lua

**Reply by cscheid** [reply]
Thanks for spotting that. I think the right fix is somehow for quarto to communicate the information from the JSON file to Lua instead of replicating the data there (we try hard to adhere to the principle of single source of truth). Unfortunately, that means that the fix will take more work. Can you open a new issue? Thanks.

**Reply by eitsupi** [reply]
Sorry, I looked again carefully and it seems that perhaps what exists in the lua file is a list of languages that are registered as valid syntax highlighting in Pandoc.
Since the purpose is different, the same list cannot be used, and since `prql` is not valid in Pandoc, I don't see the point in registering it.
Sorry for the trouble.

**Reply by eitsupi** [reply]
I was trying version 1.3 and noticed that code annotations do not work in prql code blocks.
I think it is because they are not included here, so we still need to create an issue as you suggested.

https://github.com/quarto-dev/quarto-cli/blob/94a815fa510272ed55843cde80ad20cac2e6c2bc/src/resources/filters/quarto-pre/code-annotation.lua#L6-L54

**Reply by eitsupi** [reply]
I created a new issue #4823
